### PR TITLE
Enable sorting of spring list

### PR DESCRIPTION
**Pull request by AnkushMalik**
Use table to arrange the sprints and made them sortable on clicking the column name

Closes https://github.com/openSUSE/agile-team-dashboard/issues/85
  
screenshot-2018-1-7 dashboard

### CODE REVIEW

**Comment by AnkushMalik** [comment]
@Ana06  can you please review this PR? :)

File: Gemfile.lock [hunk 1]
```diff
@@ -333,7 +333,6 @@ GEM
       rubocop (>= 0.47.0)
       slim (~> 3.0)
       sysexits (~> 1.1)
-    slop (3.6.0)
```
Comment (Ana06) [inline Gemfile.lock]: who is that related to this commit? 🤔

File: app/controllers/sprints_controller.rb [hunk 2]
```diff
@@ -11,8 +11,8 @@ def index
     self.params = params.permit!
     smart_listing_create :sprints, Sprint.all, partial: 'sprints/listing', default_sort: { number: 'asc' },
                                                sort_attributes: [[:number, 'number'],
-                                                                 [:end_date, 'end_date'],
-                                                                 [:start_date, 'start_date']]
+                                                                 [:start_date, 'start_date'],
```
Comment (Ana06) [inline app/controllers/sprints_controller.rb]: please squash this to the previous commit

File: app/views/sprints/_listing.html.slim [hunk 3]
```diff
@@ -13,9 +14,19 @@
             = "#{sprint.start_date.strftime("%d-%m-%Y")}"
           td.description
             = "#{sprint.end_date.strftime("%d-%m-%Y")}"
+          td.description
```
Comment (Ana06) [inline app/views/sprints/_listing.html.slim]: why do you use `description` here?

File: app/views/sprints/_listing.html.slim [hunk 4]
```diff
@@ -0,0 +1,21 @@
+- unless smart_listing.empty?
+  table.ui.celled.table
+    thead
+      th.header = smart_listing.sortable "Number", "number"
+      th.header = smart_listing.sortable "Start Date", "start_date"
+      th.header = smart_listing.sortable "End Date", "end_date"
+    tbody
+      - smart_listing.collection.each do |sprint|
+        tr
+          td.header
```
Comment (Ana06) [inline app/views/sprints/_listing.html.slim]: why do you use `header` here?

File: app/views/sprints/_listing.html.slim [hunk 5]
```diff
@@ -13,9 +14,19 @@
             = "#{sprint.start_date.strftime("%d-%m-%Y")}"
           td.description
             = "#{sprint.end_date.strftime("%d-%m-%Y")}"
+          td.description
+            span style="padding-left: 10px;"
```
Comment (Ana06) [inline app/views/sprints/_listing.html.slim]: It would be nice if your table included this as well, so a unstarted sprting 😉

File: app/views/sprints/_listing.html.slim [hunk 6]
```diff
@@ -13,9 +14,19 @@
...
+              - if sprint.unstarted_sprint?
+                = link_to sprint_start_path(sprint) do
+                  i.video.play.icon
+                    start sprint
+            span.action.right.floated
```
Comment (Ana06) [inline app/views/sprints/_listing.html.slim]: This classes are not having any effect. I would prefer to center it by the way.

File: Gemfile.lock (hunk 1 above)
Comment (AnkushMalik) [reply Gemfile.lock]: i dont know why but this line is removed whenever i do vagrant exec bundle install

File: app/views/sprints/_listing.html.slim (hunk 6 above)
Comment (AnkushMalik) [reply app/views/sprints/_listing.html.slim]: my apologies, i was ignorant. just copy pasted :sweat_smile:

File: Gemfile.lock (hunk 1 above)
Comment (Ana06) [reply Gemfile.lock]: I think it is part of your first commit, when you installed the gem it changed the dependencies in the .lock file. It is just in the wrong commit.

File: app/views/sprints/_listing.html.slim (hunk 5 above)
Comment (AnkushMalik) [reply app/views/sprints/_listing.html.slim]: I didnt get you , u mean a different column for this 'start sprint'  link?

File: app/views/sprints/_listing.html.slim (hunk 6 above)
Comment (DavidKang) [reply app/views/sprints/_listing.html.slim]: @AnkushMalik, the problem with this table is that the action button doesn't match very well, have a look at the next screenshot:
screenshot from 2018-01-10 10-14-25

I think could do something like:

screenshot from 2018-01-10 10-18-04

and if it's possible I would add the title attribute in the link_to to know what does each action button.

**Comment by AnkushMalik** [comment]
screenshot-2018-1-10 dashboard

done :)

File: app/views/sprints/_listing.html.slim [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      th.center.aligned = smart_listing.sortable "End Date", "end_date"
+      th.center.aligned Actions
+    tbody
+      - smart_listing.collection.each do |sprint|
+        tr
+          td.center.aligned = sprint.number
+          td.center.aligned = "#{sprint.start_date.strftime("%d-%m-%Y")}"
+          td.center.aligned = "#{sprint.end_date.strftime("%d-%m-%Y")}"
+          td
+            span.action.right.floated.ui.list
```
Comment (DavidKang) [inline app/views/sprints/_listing.html.slim]: @AnkushMalik, I think you don't need these classes, only adding the classes `.right.aligned` in the previous `td` should be enough.

**Comment by DavidKang** [comment]
@AnkushMalik, the reason why travis is failing is there: https://travis-ci.org/openSUSE/agile-team-dashboard/jobs/327233983#L899

@AnkushMalik  do you need help with this?

**Comment by AnkushMalik** [comment]
no its fine , error message didnt appeared last time

File: app/controllers/sprints_controller.rb [hunk 8]
```diff
@@ -4,9 +4,15 @@
 
 class SprintsController < ApplicationController
   before_action :set_sprint, only: %i[show edit update destroy start]
+  include SmartListing::Helper::ControllerExtensions
+  helper  SmartListing::Helper
 
   def index
-    @sprints = Sprint.all
+    self.params = params.permit!
```
Comment (Ana06) [inline app/controllers/sprints_controller.rb]: why is this needed?

Comment (AnkushMalik) [reply app/controllers/sprints_controller.rb]: I was having the same problem as this one :-  https://github.com/Sology/smart_listing/issues/134
I tried everything but couldnt find better solution than this one :/

Comment (AnkushMalik) [reply app/controllers/sprints_controller.rb]: I was having the same problem as this one : https://github.com/Sology/smart_listing/issues/134
I tried everthing but couldn't find any better solution than this one :/

Comment (Ana06) [reply app/controllers/sprints_controller.rb]: ok, can you please write a TODO comment in the code?

and we should create an issue so we do not forget as well.

**Comment by Ana06** [comment]
@AnkushMalik can you please open an issue so we do not forget of what is missed? 😉
